swarm/pss/pss.go

```diff
@@ -409,135 +399,211 @@ func (self *Pss) Process(pssmsg *PssMsg) error {
 }
 
 // attempt to decrypt, validate and unpack sym msg
-func (self *Pss) processSym(envelope *whisper.Envelope) (recvmsg *whisper.ReceivedMessage, frombytes []byte, err error) {
-	for i := self.symkeycachecursor; i > self.symkeycachecursor-cap(self.symkeycache) && i > 0; i-- {
-		symkeyid := self.symkeycache[i%cap(self.symkeycache)].recvsymkey
+// if successful, returns the unpacked whisper ReceivedMessage struct
+// encapsulating the decrypted message, and the symkeyid of the symkey
+// used to decrypt the message, the latter revealing the sender
+// fails if decryption of message fails or if the message is corrupted
```

Comment:
> // used to decrypt the message, the latter revealing the sender
> // fails if decryption of message fails or if the message is corrupted

used to decrypt the message, the latter revealing the sender.
It fails if decryption of message fails or if the message is corrupted